scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py

```diff
@@ -99,6 +104,11 @@ def focus_run(self, sample_paths, banks, plot_output, instrument, rb_num, spectr
                     else:
                         df_kwarg = {"GroupingFileName": SOUTH_BANK_CAL}
                         region_calib = "engggui_calibration_bank_2"
+                    # check correct region calibration exists
+                    if not Ads.doesExist(region_calib):
```

Comment:
This fixes the problem I saw but can this check go earlier? No worries if not/it is a lot of effort- I know we plan to consolidate the ROI selection into a single place in the UI so it is fine for now.

Reply:
I think to put it earlier we'd have to duplicate the check to see what region of interest is selected, and to time of execution for the code before it returns is fairly small, so in my opinion I think this is best left for the PR to include the single ROI selection